src/lib/components/WebvizPluginPlaceholder.react.js

```diff
@@ -33,13 +33,9 @@ export default class WebvizPluginPlaceholder extends Component {
     }
 
     componentDidUpdate(_, prevState) {
-        if (this.props.zip_base64 !== "") {
-            const now = new Date();
-            const filename = `webviz-data-${now.getFullYear()}-${now.getMonth() +
-                1}-${now.getDate()}.zip`;
-
-            download_file(filename, this.props.zip_base64);
-            this.props.setProps({ zip_base64: "" });
+        if (JSON.stringify(this.props.download) !== "{}" && this.props.download !== "") {
```

Comment:
There is a need for checking that is not an empty string is you do not want to introduce a breaking change in the way `` the `webviz-config` plugin was implemented by consumers currently this is the recommended way:

```python
@app.callback(self.plugin_data_output,
              [self.plugin_data_requested])
def _user_download_data(data_requested):
    return WebvizPluginABC.plugin_data_compress(
        [{'filename': 'some_file.txt',
          'content': 'Some download data'}]
    ) if data_requested else ''
```

we have also used this way in `everviz` and without checking for empty string we will be forced to update the implementation it is not a big change but it will be required (not really sure how many `webviz-config` consumers are out there)

Reply:
> There is a need for checking that is not an empty string is you do not want to introduce a breaking change.

Good point. Maybe we can for now add a sentence here that we `stringify`, and also compare against empty string, because we will go into an deprecation phase for `""`.